packages/teleport/src/DesktopSession/useTdpClientCanvas.tsx

```diff
@@ -224,9 +229,12 @@ export default function useTdpClientCanvas(props: Props) {
   // on the remote machine.
   const onContextMenu = () => false;
 
-  const sendLocalClipboardToRemote = (cli: TdpClient) => {
-    // We must check that the DOM is focused or navigator.clipboard.readText throws an error.
-    if (clipboardSharingEnabled && document.hasFocus()) {
+  const sendLocalClipboardToRemote = async (cli: TdpClient) => {
+    if (
+      clipboardSharingEnabled &&
+      document.hasFocus() &&
+      (await shouldTryClipboardRW())
```

Comment:
I think you can move the whole condition to a function and reuse it in both places.